wsmaster/che-core-api-auth-bitbucket/src/main/java/org/eclipse/che/security/oauth/BitbucketOAuthAuthenticator.java

```diff
@@ -52,7 +54,7 @@ public String getAuthenticateUrl(URL requestUrl, List<String> scopes) {
 
   @Override
   public final String getOAuthProvider() {
-    return "bitbucket";
+    return BITBUCKET_CLOUD_ENDPOINT.equals(bitbucketEndpoint) ? "bitbucket" : "bitbucket-server";
```

Comment:
should we extract consts as well for bitbucket / bitbucket-server ? would be nice to add comments java docs as well

Reply:
done